src/rdkafka_txnmgr.c

```diff
@@ -2086,14 +2174,26 @@ static void rd_kafka_txn_handle_EndTxn(rd_kafka_t *rk,
 err:
         rd_kafka_wrlock(rk);
 
-        if (rk->rk_eos.txn_state == RD_KAFKA_TXN_STATE_COMMITTING_TRANSACTION) {
+        int state_committing =
```

Comment:
these booleans do make it more readable IMO (good idea!), however:
1. you need to declare the variables at the top of the function, not here.
2. i don't think it's consistent with the librdkafka coding style to use them - just use the == expressions (even though it'll be harder to read)

Reply:
actually I take back point #2 - the logic for `state_different_from_request` really does benefit enough to keep them...